woocommerce/woocommerce-android
# [REST API] Finalize Jetpack activation for application password case and logs user in

Closes #8697 

After Jetpack activation (installation and/or connection) is completed, previously the app refresh the state of the app. This PR completes the activation process by setting the site properly, and removing all previous sites in the DB that doesn't use WPCom REST API.

In effect, after activation is completed, this returns the merchant back to the "My Store" screen, but it's now in the Jetpack-connected state where every features are now available.

### To test:
1. Prepare a test self-hosted site with no Jetpack plugin (or Jetpack plugin active, but the account disconnected),
2. Start app, login and complete the site credentials flow, this logs in the user in the Application Password flow,
3. on "My Store" screen, tap the Jetpack banner at the bottom,
4. on the Jetpack Benefit screen that appears, click the "Log In to Continue" button,
5. Go through the Jetpack installation / connection flow, until everything is completed and the "Continue" button is shown,
6. Tap "Continue", make sure you're now logged in to "My Store" in the WordPress.com login flow,
7. Test around the app especially for features that's only available on WordPress.com flow:*
   1. Visitor stats,
   2. Switching store if you have more than one in the same WordPress.com account,
   3. Notifications
